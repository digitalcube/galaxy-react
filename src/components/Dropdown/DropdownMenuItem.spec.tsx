/* eslint-disable jsx-a11y/anchor-is-valid */
import { render } from '@testing-library/react';
import React from 'react';
import { FaTrash } from 'react-icons/fa';
import { DropdownMenuItem } from './DropdownMenuItem';

describe('DropdownMenuItem', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <DropdownMenuItem>
        <a href="#">Item</a>
      </DropdownMenuItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot [full props]', () => {
    const { asFragment } = render(
      <DropdownMenuItem
        right
        icon={<FaTrash className="d-block" />}
        className="test"
      >
        <a href="#">Item</a>
      </DropdownMenuItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
