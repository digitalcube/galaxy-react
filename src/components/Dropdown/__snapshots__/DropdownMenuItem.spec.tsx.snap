// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenuItem to match snapshot [full props] 1`] = `
<DocumentFragment>
  <button
    class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between test dropdown-item"
    role="menuitem"
    tabindex="0"
    type="button"
  >
    <a
      href="#"
    >
      Item
    </a>
    <i
      aria-hidden="true"
    >
      <svg
        class="d-block"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
        />
      </svg>
    </i>
  </button>
</DocumentFragment>
`;

exports[`DropdownMenuItem to match snapshot 1`] = `
<DocumentFragment>
  <button
    class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between dropdown-item"
    role="menuitem"
    tabindex="0"
    type="button"
  >
    <a
      href="#"
    >
      Item
    </a>
  </button>
</DocumentFragment>
`;
